lib/SILOptimizer/Transforms/AccessEnforcementOpts.cpp

```diff
@@ -433,37 +436,26 @@ void AccessConflictAndMergeAnalysis::addOutOfScopeAccess(
     }
   }
 
-  info.outOfScopeConflictFreeAccesses.conflictFreeAccesses.insert(
-      std::make_pair(newIndex, beginAccess));
-}
-
-static void copyMap(DenseAccessMap &to, const DenseAccessMap &from) {
-  for (auto it : from) {
-    to.insert(it);
-  }
+  info.outOfScopeConflictFreeAccesses.conflictFreeAccesses.insert(beginAccess);
 }
 
 void AccessConflictAndMergeAnalysis::mergeAccessStruct(
     RegionInfo &info, RegionInfo::AccessSummary &accessStruct,
     const RegionInfo::AccessSummary &RHSAccessStruct) {
   if (!accessStruct.merged) {
-    copyMap(accessStruct.conflictFreeAccesses,
-            RHSAccessStruct.conflictFreeAccesses);
+    accessStruct.conflictFreeAccesses.insert(
+        RHSAccessStruct.conflictFreeAccesses.begin(),
+        RHSAccessStruct.conflictFreeAccesses.end());
     accessStruct.merged = true;
     return;
   }
 
-  DenseAccessMap tmpMap;
-  copyMap(tmpMap, accessStruct.conflictFreeAccesses);
-
-  for (auto pair : tmpMap) {
-    auto index = pair.first;
-    if (RHSAccessStruct.conflictFreeAccesses.find(index) !=
-        RHSAccessStruct.conflictFreeAccesses.end())
-      continue;
-    // Not in RHS - remove from intersect
-    accessStruct.conflictFreeAccesses.erase(index);
-  }
+  auto pred = [&](BeginAccessInst *it) {
+    auto rhsIt = std::find(RHSAccessStruct.conflictFreeAccesses.begin(),
```

Comment:
Checking Set membership is done with `count`.

Reply:
Fixed - forgot to change that